canonical-web-and-design/maas-ui
# feat(ui): add DHCP edit form to the network tab

## Done

- Move the DHCP form to base/components and update to be reusable outside of settings.
- Add the form to edit DHCP snippets in the DHCP table in machine details.

## QA

### MAAS deployment

To run this branch you will need access to one of the following MAAS deployments:

- [Bolla](/HACKING.md#bolla)
- [A development MAAS](/HACKING.md#development-deployment)
- [A local snap MAAS](/HACKING.md#snap-deployment) (this will not usually have machines)

### Running the branch

You can run this branch by:

- Serving with [dotrun](/HACKING.md#maas-ui-development-setup)
- [Building in a development MAAS](/HACKING.md#running-maas-ui-from-a-development-maas)

### QA steps

- Go to the DHCP settings (/MAAS/r/settings/dhcp).
- Add a snippet and set the type to a machine.
- Go to the new network tab for the machine you added the DHCP snippet to.
- In the DHCP table click the pencil, it should open the edit form.
- Change something and save, the form should close.

## Fixes

Fixes: #1890.
